frontend/app/routes/protected/person-case/primary-docs.tsx

```diff
@@ -0,0 +1,110 @@
+      currentStatusInCanada: context.session.inPersonSINCase?.currentStatusInCanada,
+    },
+  };
+}
+
+export const meta: Route.MetaFunction = ({ data }) => {
+  return [{ title: data.documentTitle }];
+};
+
+export async function action({ context, request }: Route.ActionArgs) {
+  requireAuth(context.session, new URL(request.url), ['user']);
+  const lang = getLanguage(request);
+
+  const formData = await request.formData();
+  const input = { currentStatusInCanada: formData.get('currentStatusInCanada') as string };
+
+  const schema = v.object({ currentStatusInCanada: currentStatusInCanadaSchema() });
+  const parsedDataResult = v.safeParse(schema, input, { lang });
+
+  if (!parsedDataResult.success) {
+    return data({ errors: v.flatten<typeof schema>(parsedDataResult.issues).nested }, { status: 400 });
+  }
+
+  context.session.inPersonSINCase = {
+    ...(context.session.inPersonSINCase ?? {}),
+    ...input,
+  };
+  return i18nRedirect('routes/protected/person-case/first-name.tsx', request); //TODO: change it to redirect to file="routes/protected/person-case/secondary-docs.tsx"
+}
+
+export default function PrimaryDocs({ loaderData, actionData, params }: Route.ComponentProps) {
+  const { t } = useTranslation(handle.i18nNamespace);
+
+  const fetcher = useFetcher<Info['actionData']>();
+  const isSubmitting = fetcher.state !== 'idle';
+  const errors = fetcher.data?.errors;
+
+  return (
+    <>
+      <Progress className="mt-8" label="" value={30} />
+      <PageTitle subTitle={t('protected:in-person.title')}>{t('protected:primary-identity-document.page-title')}</PageTitle>
+
+      <fetcher.Form method="post" noValidate>
+        <div className="space-y-6">
+          <ErrorSummary errors={errors} />
+
+          <InputSelect
+            id="currentStatusInCanada"
+            name="currentStatusInCanada"
+            errorMessage={errors?.currentStatusInCanada?.[0]}
+            defaultValue={loaderData.defaultFormValues.currentStatusInCanada}
+            required
+            options={[
+              {
+                children: t('protected:primary-identity-document.current-status-in-canada.option-1'),
+                value: 'canadianCitizenBornOutsideCanada',
+              },
+            ]}
+            label={t('protected:primary-identity-document.current-status-in-canada.title')}
+          />
+        </div>
+        <div className="mt-8 flex flex-wrap items-center gap-3">
+          <ButtonLink
+            id="back-button"
+            file="routes/protected/person-case/privacy-statement.tsx" //TODO: change it to redirect to file="routes/protected/person-case/request-details.tsx"
+            params={params}
+            disabled={isSubmitting}
+          >
+            {t('protected:person-case.previous')}
+          </ButtonLink>
+          <Button variant="primary" type="submit" id="continue-button" disabled={isSubmitting}>
+            {t('protected:person-case.next')}
+          </Button>
```

Comment:
Since this entire flow will eventually be handled by a finite state machine, the next and previous buttons should both be buttons that submit. Something like this:

``` tsx
<Button name="action" value="back" variant="secondary">Previous</Button>
<Button name="action" value="next" variant="primary">Next</Button>
```

Then, to handle these in the `action()` function, you would do something like this:

``` typescript
switch (formData.get('action')) {
  case 'back': {
    // handle 'previous' button click
  }
  case 'next': {
    // handle 'next' button click
  }
}
```

Reply:
Also.. Notice that I didn't use `type="submit"`. This is because for a button, `submit` is the default type, so it can be omitted.